bundles/org.openhab.core/src/main/java/org/openhab/core/internal/items/MetadataRegistryImpl.java

```diff
@@ -61,6 +64,18 @@ public boolean isInternalNamespace(String namespace) {
         return namespace.startsWith(INTERNAL_NAMESPACE_PREFIX);
     }
 
+    /**
+     * Provides all namespaces of a particular item
+     *
+     * @param itemname the name of the item for which the namespaces should be searched.
+     */
+    @Override
+    public Set<String> getAllNamespaces(String itemname) {
+        return getAll().stream().filter(n -> {
+            return n.getUID().getItemName().equals(itemname);
+        }).map(n -> n.getUID().getNamespace()).collect(Collectors.toSet());
```

Comment:
```suggestion
    public Collection<String> getAllNamespaces(String itemname) {
        return stream().map(Metadata::getUID).filter(key -> key.getItemName().equals(itemname)).map(MetadataKey::getNamespace).toList();
```

You can use `stream()` from `AbstractRegistry` and by mapping to the `UID` (a `MetadataKey`) first also shorten the further processing.

Reply:
The reason only becomes obvious if you see the implementation of the UI. The way at least how the UI works currently is that it "knows" what the standard metadata-Namespaces are. Basically what it does, that it explicitly asks for the namespace it knows because that is what the API requires (this API here does not deliver any metadata, if you not explicitly ask for each individual namespace). 
What I basically did in the UI now, is that I ask for all names and exclude the ones that the UI already knows and then add them to the list the UI wants. Basically, and I agree here, will most likely result in the full set again. The only reason I did that was that I wasn't sure if @ghys only did this because it wasn't possible otherwise (which most likely is the case) or if he had something else in mind as well.

To summarize: I could easily take the exclude parameter out and ask for all namespaces. Worst case that could habe here that would have to do an exclusion in the frontend. So, let me know if I should delete the exclusion part. I would do so and simplify the UI to just take all namespaces it gets from the backend API.

While thinking about it, we could even go a bit further and make getItems() smart by providing the _metadata_-parameter with something like "*" which would internally grap all namespaces and deliver them automatically which would avoid one round-trip (setting _metadata_ to "" will return nothing). What about that idea?

Reply:
It's easy to filter a list on another list in js and also to concatenate them to a string:

```
let customNamespaces = allNamespaces.filter(ns => !knownNamespaces.includes(ns)).join(',')
```

and returning the list of namespaces as JSON array makes it much easier to re-use for other use-cases. So I would suggest to change to `@Produces(MediaType.APPLICATION_JSON)` and use `Response.ok(new Stream2JSONInputStream(namespaces.stream())).build();`

Regarding the change in `.getItems`: I never understood why it was designed that way. IMO it would have been much better to return everything by default and allow to limit it with the parameter than the other way round. @ghys, @lolodomo Would it hurt for UI to change that?

Reply:
"I never understood why it was designed that way. IMO it would have been much better to return everything by default and allow to limit it with the parameter than the other way round."

At least it would break the current behaviour. That's why I didn't dare to do it. On the other hand, why would I want to deliver all the namespace information in getItems when most of the time it is not needed. Therefore I do understand the design.

Ok, let's summarise (as I want to get this thing done eventually):
- I remove the exclusion parameter
- I change the return parameter to be JSON
- I adapt the UI accordingly
- I keep the getItem as it is for now.

Reply:
Perfectly summed up.

Reply:
> I never understood why it was designed that way. IMO it would have been much better to return everything by default and allow to limit it with the parameter than the other way round.

Not sure, but maybe performance reasons? @kaikreuzer designed this originally and might have the answer. It would probably be wise to have this answer before making changes that are harmful.

> @ghys, @lolodomo Would it hurt for UI to change that?

Probably not, it would simply return more metadata than necessary.

Note that the idea I had over 2 years ago but never implemented (https://github.com/openhab/openhab-webui/issues/397#issuecomment-717130372) to solve this user namespaces listing issue without changing the API was to have a managed [MetadataConfigDescriptionProvider](https://next.openhab.org/javadoc/latest/org/openhab/core/config/core/metadata/metadataconfigdescriptionprovider) backed by UI components (like UI-designed sitemaps, though it can have its own custom storage/syntax), along with an editor in the UI's developer tools. Then MetadataConfigDescriptionProviders declared this way along with others e.g. coming from add-ons could appear in the list (the "enter custom namespace" option would still be there). When editing metadata for a described namespace the user would be presented with a nice standard configuration sheet with parameters, not only the a raw YAML editor.
I saw this recently on a forum thread too and it probably could make it to 4.0.

Reply:
@ghys did you see my contribution here?
https://github.com/openhab/openhab-webui/pull/1623